[VDG] [Fluent] Allow Coinjoins to be expanded in the TreeDataGrid

wieslawsoltes opened the pull request.

MaxHillebrand reviewed (commented):
tested [4f1bdf6](https://github.com/zkSNACKs/WalletWasabi/pull/7510/commits/4f1bdf69cfe6780d5c29bc83d382f2867dccc06d)

expand and collaps works well. cj tx details works. 

When expanded, the included coinjoin has a tx icon, not the shield icon. 

awesome work, this is really promissing!

yahiheb commented:
Tested. This looks good.

One thing I noticed that can maybe improve is that if you have only one coinjoin tx then there is no need to expand it.

wieslawsoltes commented:
> One thing I noticed that can maybe improve is that if you have only one coinjoin tx then there is no need to expand it.

Will look into that. Should be possibile.

wieslawsoltes commented:
> When expanded, the included coinjoin has a tx icon, not the shield icon.

@MaxHillebrand fixed icon

wieslawsoltes commented:
> Tested. This looks good.
> 
> One thing I noticed that can maybe improve is that if you have only one coinjoin tx then there is no need to expand it.

@yahiheb added check and now only if more then one transaction is present expander is visible

yahiheb commented on WalletWasabi.Fluent/ViewModels/Wallets/Home/History/HistoryItems/CoinJoinHistoryItemViewModel.cs:
```diff
@@ -0,0 +1,45 @@
+		IObservable<Unit> updateTrigger)
+		: base(orderIndex, transactionSummary)
+	{
+		Label = "Coinjoin";
```
@soosr  Why do we label the coinjoin transactions?

soosr replied:
It was `Privacy boost` previously. It is ok for me to remove the redundant `Coinjoin` text.

yahiheb replied:
My question is more like why do we need the label for coinjoin txs?

wieslawsoltes replied:
Remove label text from this PR

yahiheb replied:
IMO the label in this PR was good because it is only one coinjoin.

I am just curious to know why do we even need the label for coinjoin txs.

soosr replied:
> I am just curious to know why do we even need the label for coinjoin txs.

We don't and I don't remember why we had that. Maybe it was an early decision and we got used to it and stick to that.

Now I don't think we need that. There is the type shield that indicates that it is a CJ transaction.
There were talks about adding the number of coinjoins into the label, but it seems that is not needed.

And also in the future, we want to show the `Recipient` in the history rather than all the entity that knows about it. `Coinjoin` text would not fit there.

yahiheb reviewed (approved):
tACK

MaxHillebrand commented:
works great!
Maybe it can be indented a bit more, it's a weird little amount now. Maybe alternatively make it a different background color?
![2022-03-21-084318](https://user-images.githubusercontent.com/30683012/159221135-78513886-3410-4dc3-9f49-f791870231a9.png)

wieslawsoltes commented:
> works great! Maybe it can be indented a bit more, it's a weird little amount now. Maybe alternatively make it a different background color? ![2022-03-21-084318](https://user-images.githubusercontent.com/30683012/159221135-78513886-3410-4dc3-9f49-f791870231a9.png)

Changed indentation and added different background color.

MaxHillebrand reviewed (dismissed):
tACK [743cc71](https://github.com/zkSNACKs/WalletWasabi/pull/7510/commits/743cc71c3ca3f7d48ade0c853fe9247c11e3bf00)
![2022-03-22-122116](https://user-images.githubusercontent.com/30683012/159470683-ec088a5c-c055-48bc-9564-6be48f1c90b5.png)

yahiheb reviewed (dismissed):
tACK

yahiheb commented on WalletWasabi.Fluent/ViewModels/Wallets/Home/History/HistoryItems/CoinJoinHistoryItemViewModel.cs:
```diff
@@ -0,0 +1,47 @@
+		IObservable<Unit> updateTrigger)
+		: base(orderIndex, transactionSummary)
+	{
+		Label = transactionSummary.Label.Take(1).FirstOrDefault();
```
We might not need this line if #7604 gets accepted.

soosr commented on WalletWasabi.Fluent/ViewModels/Wallets/Home/History/HistoryItems/HistoryItemViewModelBase.cs:
```diff
@@ -47,6 +52,21 @@ protected HistoryItemViewModelBase(int orderIndex, TransactionSummary transactio
+		throw new NotSupportedException();
+	}
+
+	public virtual void Update(HistoryItemViewModelBase item)
+	{
+		OrderIndex = item.OrderIndex;
+		Date = item.Date;
+		DateString = item.DateString;
+		IsConfirmed = item.IsConfirmed;
+	}
+
+	public bool IsSimilar(HistoryItemViewModelBase item) => Id == item.Id;
```
They are removed on master. You can do as well

wieslawsoltes replied:
It got added accidentally after my last merge resolve conflict

wieslawsoltes replied:
Removed

soosr reviewed (commented):
nit

MaxHillebrand reviewed (approved):
tACK [559bae6](https://github.com/zkSNACKs/WalletWasabi/pull/7510/commits/559bae6fe09dc2f5e346ae41d95b93f1030fd67b)

soosr reviewed (approved):
LGTM

MaxHillebrand commented:
There is no change in the mouse icon when hovering over the expand arrow, so it's not obvious that it's click able.

![IMG_20220404_132642.jpg](https://user-images.githubusercontent.com/30683012/161534351-367b36b2-0f4d-49e6-b6c9-c1ccf92efeb7.jpg)![IMG_20220404_132603.jpg](https://user-images.githubusercontent.com/30683012/161534360-1f4d4fce-9c85-4e2c-b216-c60197acbdd0.jpg)

wieslawsoltes commented:
> There is no change in the mouse icon when hovering over the expand arrow, so it's not obvious that it's click able.
> 
> ![IMG_20220404_132642.jpg](https://user-images.githubusercontent.com/30683012/161534351-367b36b2-0f4d-49e6-b6c9-c1ccf92efeb7.jpg)![IMG_20220404_132603.jpg](https://user-images.githubusercontent.com/30683012/161534360-1f4d4fce-9c85-4e2c-b216-c60197acbdd0.jpg)

@MaxHillebrand 
https://github.com/zkSNACKs/WalletWasabi/pull/7695
